File: Baekjoon/notUploaded/7573.js
const reader = {
    input: require('fs').readFileSync('input.txt').toString().split("\r\n"),
    // input : require('fs').readFileSync('/dev/stdin').toString().split("\n"),
    index: 0,
    read(){
        return this.input[this.index++];
    }
}
let [N,I,M] = reader.read().split(" ").map(e => parseInt(e));
I /= 2;
const Fishs = new Set();
let answer = 0;
for (let i=0; i<M; i++)
    Fishs.add(reader.read());
for (let [R,C] of Array.from(Fishs).map(e=>e.split(' ').map(e=>parseInt(e)))){
    for (let height = 1; height<I; height++){
        let width = I-height;
        let count = 0;
        for (let r = R+height; r>=R; r--){
            for (let c = C+width; c>=C; c--){
                if (Fishs.has(r+" "+c))
                    count+=1;
            }
        }
        if (count > answer)
            answer = count;
    }
}
console.log(answer);

File: Baekjoon/problem/python/7573.py
import sys;

N,I,M = map(int, sys.stdin.readline().split());
I = int(I/2);
scanList = set();

def apply(a,b, I):
    for h in range(1,I):
        w = I-h;
        for i in range(h+1):
            scanList.add((a-i, b, h,w));
    return;

fishs = set();
for _ in range(M):
    a,b = map(int, sys.stdin.readline().split());
    fishs.add((a,b));
    apply(a,b, I);
res = 0;
for a,b,c,d in scanList:
    count = 0;
    if a < 0 or b < 0:
        continue;
    for i in range(c+1):
        for j in range(d+1):
            if (a+i,b+j) in fishs:
                count += 1;
    if count > res:
        res = count;
print(res);